test/Microsoft.ML.AutoML.Tests/DatasetUtil.cs

```diff
@@ -35,6 +37,126 @@ public static IDataView GetUciAdultDataView()
+                folder: fullImagesetFolderPath, useFolderNameAsLabel: true);
+
+
+            using (StreamWriter file = new StreamWriter(datasetName))
+            {
+                file.WriteLine("Label,ImagePath");
+                foreach (var image in images)
+                    file.WriteLine(image.Label + "," + image.ImagePath);
+            }
+
+            return datasetName;
+        }
+
+        public static IEnumerable<ImageData> LoadImagesFromDirectory(string folder,
+            bool useFolderNameAsLabel = true)
+        {
+            var files = Directory.GetFiles(folder, "*",
+                searchOption: SearchOption.AllDirectories);
+            foreach (var file in files)
+            {
+                if (Path.GetExtension(file) != ".jpg")
+                    continue;
+
+                var label = Path.GetFileName(file);
+                if (useFolderNameAsLabel)
+                    label = Directory.GetParent(file).Name;
+                else
+                {
+                    for (int index = 0; index < label.Length; index++)
+                    {
+                        if (!char.IsLetter(label[index]))
+                        {
+                            label = label.Substring(0, index);
+                            break;
+                        }
+                    }
+                }
+
+                yield return new ImageData()
+                {
+                    ImagePath = file,
+                    Label = label
+                };
+
+            }
+        }
+
+        public static string DownloadImageSet(string imagesDownloadFolder)
+        {
+            string fileName = "flower_photos_tiny_set_for_unit_tests.zip";
+            string url = $"https://mlnetfilestorage.file.core.windows.net/imagesets" +
+                $"/flower_images/flower_photos_tiny_set_for_unit_tests.zip?st=2019" +
+                $"-08-29T00%3A07%3A21Z&se=2030-08-30T00%3A07%3A00Z&sp=rl&sv=2018" +
+                $"-03-28&sr=f&sig=N8HbLziTcT61kstprNLmn%2BDC0JoMrNwo6yRWb3hLLag%3D";
+
+            Download(url, imagesDownloadFolder, fileName);
+            UnZip(Path.Combine(imagesDownloadFolder, fileName), imagesDownloadFolder);
+
+            return Path.GetFileNameWithoutExtension(fileName);
+        }
+
+        private static bool Download(string url, string destDir, string destFileName)
+        {
+            if (destFileName == null)
+                destFileName = url.Split(Path.DirectorySeparatorChar).Last();
+
+            Directory.CreateDirectory(destDir);
+
+            string relativeFilePath = Path.Combine(destDir, destFileName);
+
+            if (File.Exists(relativeFilePath))
+                return false;
+
+            new WebClient().DownloadFile(url, relativeFilePath);
+            return true;
+        }
+
+        private static void UnZip(String gzArchiveName, String destFolder)
+        {
+            var flag = gzArchiveName.Split(Path.DirectorySeparatorChar)
```

Comment:
Path. GetFileNameWithoutExtension is made for this. 
Instead of `gzArchiveName.Split(Path.DirectorySeparatorChar).Last().Split('.').First() + ".bin"`
```suggestion
            var flag = Path.GetFileNameWithoutExtension(gzArchiveName) + ".bin"
```